### PR TITLE
AppVeyor CI: check if tests not fail

### PR DESCRIPTION
**Pull request by chfast**
Fixes testing with EVMJIT.
Resolves https://github.com/ethereum/cpp-ethereum/issues/3039.

### CODE REVIEW

File: scripts/tests.bat [hunk 1]
```diff
@@ -31,7 +31,7 @@ set ETHEREUM_DEPS_PATH=%4
 if "%TESTS%"=="On" (
 
     cd ..
-    git clone https://github.com/ethereum/tests.git
+    git clone --depth 1 https://github.com/ethereum/tests.git
```
Comment (bobsummerwill) [inline scripts/tests.bat]: Why do we need `--depth 1` here?   Are there sub-modules?   I don't see any.

**Comment by bobsummerwill** [comment]
This PR could probably also benefit from a refresh from `develop`.

But yeah - looks like there are probably real unit-test failures here.    I wonder if they are specific to `evmjit`?    Could you maybe try seeing if they a replicable against the Interpreter too?    Either by tweaking this PR for a cycle or two, or by pulling just the error-code handling stuff out into its own PR?   Thanks!

File: scripts/tests.bat (hunk 1 above)
Comment (chfast) [inline scripts/tests.bat]: This is about history depth. It just downloads more/less latest commit. Nothing but optimization.

**Comment by chfast** [comment]
It is not about evmjit -- all tests within `testeth` pass (that requires evmjit.dll). What fails is `testweb3core` and `testweb3`.

**Comment by chfast** [comment]
@bobsummerwill Can we merge it?

**Comment by bobsummerwill** [comment]
We need to do something about the failing tests before enabling this.
Or all the Appveyor builds will break.

The fact that we see the issues is good, but we would just be punching ourselves in the face to turn this on without addressing those breaks.

Maybe make a new issue for each of those two breaks, detailing them?

Do you have a PC?   Maybe you can see if you can replicate them?

**Comment by bobsummerwill** [comment]
Please could you refresh your branch from `develop`, and we'll see where we are on this?
We likely need to make new issues for the Appveyor breaks, and then disable those specific tests, or fix the issues, and then we can commit this.

**Comment by bobsummerwill** [comment]
"Failed"..   Hmm.

Any idea what is happening here, @chfast?

@winsvega is also close to getting Appveyor driven windows testing enabled for Solidity.

It would be great if we could get both "on" this week, so we have correct and working quality gates on both projects.

See https://github.com/ethereum/solidity/pull/948 for that.
See

**Comment by chfast** [comment]
Yep, testeth crashes at startup: https://github.com/ethereum/cpp-ethereum/pull/3045/files#diff-289df8186113e294030c4fc9d278bbdeR48

So it is much worse that it was when I started it.
Someone would need to check that manually on Windows. What you cannot see on AppVeyor is the popup windows explaining the problem (like "cannot find X DLL").

**Comment by bobsummerwill** [comment]
Do you have a Windows setup you could check this with, please, @winsvega?

**Comment by bobsummerwill** [comment]
I believe this is redundant now.

All the changes made their way into `develop` via other PRs, with the exception of updating `cpp-dependencies`, which I would suggest we defer until https://github.com/ethereum/cpp-dependencies/pull/9 and https://github.com/ethereum/cpp-dependencies/pull/3 are resolved.

**Pull request closed without merging**